### PR TITLE
Gitlab update comment

### PR DESCRIPTION
**Pull request by dennis-sig**

### CODE REVIEW

File: test/test_gitlab_pull_request_report.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    def __init__(self, existingCommentId):
+        super().__init__()
+        self.existingCommentId = existingCommentId
+
+    def buildPostCommentURL(self, existingCommentId):
+        return str(existingCommentId)
+
+    def callAPI(self, method, url, body):
+        return self
```
Comment (patveck) [inline test/test_gitlab_pull_request_report.py:71]: Suggestion: make `callAPI` store the method (and maybe also the URL) in a class-level variable. That way, the test can verify an expectation, for instance verify that `PUT` was used in some cases and `POST` in others.

**Review by patveck — CHANGES_REQUESTED** [review]

File: test/test_gitlab_pull_request_report.py (hunk 1 above)
Comment (dennis-sig) [reply test/test_gitlab_pull_request_report.py:82]: Good suggestion, extended the tests.

File: test/test_api_caller.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        api.handleError(error)
+
+        expectedLog = [
+            "HTTP Error 404: No reason",
+            "No response headers",
+            "No response body"
+        ]
+
+        self.assertEqual(expectedLog, UploadLog.history)
+
```
Comment (patveck) [inline test/test_api_caller.py:65]: Do we want to move the tests for the other error conditions here as well (`TimeOutError`, `URLerror`, etc.)?

Comment (dennis-sig) [reply test/test_api_caller.py:66]: Fair point, moved the tests.

**Review by patveck — APPROVED** [review]

**Pull request merged**